DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -755,9 +778,15 @@ private void HandleOptionsRequest(string playerName, int options)
             if (color < 0 || color > MPColors.Count)
                 return;
 
-            var disallowedSides = GetDisallowedSides();
+            var disallowedSides = GetDisallowedSides().ToList();
+
+            // Disallowed sides from client, maps, or game modes do not take random selectors into account
+            // So, we need to insert "false" for each random at the beginning of this list AFTER getting them
+            // from client, maps, or game modes.
+            for(int i = 0; i < RandomSelectorCount; i++)
+                disallowedSides.Insert(0, false);
```

Comment:
WTF?

Reply:
https://github.com/CnCNet/xna-cncnet-client/pull/466

no idea if it is a good way or not. perhaps we should look at the original issue again